pallets/ah-migrator/src/referenda.rs

```diff
@@ -221,14 +221,20 @@ impl<T: Config> Pallet<T> {
 			count: 1,
 		});
 
-		ReferendumCount::<T, ()>::put(referendum_count);
-		deciding_count.iter().for_each(|(track_id, count)| {
-			DecidingCount::<T, ()>::insert(track_id, count);
-		});
-		track_queue.into_iter().for_each(|(track_id, queue)| {
-			let queue = BoundedVec::<_, T::MaxQueued>::defensive_truncate_from(queue);
-			TrackQueue::<T, ()>::insert(track_id, queue);
-		});
+		if let Some(referendum_count) = referendum_count {
+			ReferendumCount::<T, ()>::put(referendum_count);
+		}
+		if deciding_count.len() > 0 {
+			deciding_count.iter().for_each(|(track_id, count)| {
```

Comment:
Why the length check? If its empty then just nothing happens anyway?

Reply:
Yes, the `DecidingCount` is `ValueQuery`, no reason to insert empty vec, it can only rewrite a previous correct value.